fuzzer/src/main.rs

```diff
@@ -278,50 +282,46 @@ fn test_catch_unwind(pattern: &str) -> Result<PatternResult, ()> {
 /// This function prints its results to stdout.
 /// No further handling is needed, the returned score is for debugging purposes mostly.
 fn test(pattern: &str) -> PatternResult {
-    let mut time_samples = [(0.0, 0.0); SAMPLE_SIZE * 2];
-    // first pass, gets rid of most linear patterns if there aren't too many large outliers
-    let res1 = test_pattern(pattern, &mut time_samples[..SAMPLE_SIZE], false);
-    if let PatternResult::Linear(_) = res1 {
-        return res1;
-    }
-    // If the first pass indicated possible non-linear behaviour, retest it with a larger
-    // sample size and handle outliers.
-    let res2 = test_pattern(pattern, &mut time_samples, true);
-    if let PatternResult::Linear(_) = res2 {
-        return res2;
+    let mut time_samples = [(0.0, 0.0); SAMPLE_SIZE];
+    let mut res = PatternResult::TooLong;
```

Comment:
I don't think you need to initialize this variable, because it's always initialized before its first use.

Reply:
So I thought as well, but apparently the compiler can't infer that the loop will execute at least once..

Reply:
Oh right, the problem here is the if-condition after the for-loop, because the compiler doesn't (yet) know if the loop will ever execute, which may result in an uninitialized variable being used in the if-condition.